### PR TITLE
Improves/Reworks most tool abnormalities

### PR DESCRIPTION
**Pull request by EgorDinamit**
## About The Pull Request

- Exactly what it says in the title - all tool abnormality code has been reworked and improved.
- Skin prophecy will now additionally kill whoever goes insane.
- Theresia cannot be deactivated manually when it's doing the harmful effect; It will do it on its own after some time passes, instead.

## Why It's Good For The Game

- Good clean code is always good.
- It serves only one purpose - fear damage to those around. No longer this tool is free escape from insanity shenanigans(unless you go insane alone).
- Theresia was a really weird tool in that it was just constantly spammed to no end. Its harmful effect was also a joke(10 white damage? Seriously?). This is no longer the case.

### CODE REVIEW

File: code/modules/mob/living/simple_animal/abnormality/_tools/teth/skin.dm [hunk 1]
```diff
@@ -7,32 +7,44 @@
...
+
+// Status Effect
 /datum/status_effect/skin
 	id = "skin"
 	status_type = STATUS_EFFECT_UNIQUE
 	duration = -1
 	alert_type = null
 
-/datum/status_effect/skin/tick()
+/datum/status_effect/behavior/on_apply()
```
Comment (dsmith328) [inline code/modules/mob/living/simple_animal/abnormality/_tools/teth/skin.dm:32]: I believe this is a mistype and should be `/skin/on_apply()`

File: code/modules/mob/living/simple_animal/abnormality/_tools/teth/theresia.dm [hunk 2]
```diff
@@ -1,51 +1,74 @@
...
+
+	// Dangerous effect
+	if(current_winds >= max_winds)
+		if(activated == ACTIVE_DANGEROUS)
+			Deactivate()
+			return
+		playsound(get_turf(src), 'sound/machines/clockcult/ark_scream.ogg', 50, 0, 16)
+		new /obj/effect/temp_visual/fragment_song(get_turf(src))
+		for(var/mob/living/carbon/human/H in livinginrange(32, get_turf(src)))
+			H.apply_damage(pulse_damage, WHITE_DAMAGE, null, H.run_armor_check(null, WHITE_DAMAGE))
```
Comment (dsmith328) [inline code/modules/mob/living/simple_animal/abnormality/_tools/teth/theresia.dm:42]: This hits everyone in a 32 radius once, right?

File: code/modules/mob/living/simple_animal/abnormality/_tools/zayin/wishwell.dm [hunk 3]
```diff
@@ -128,123 +125,121 @@
...
+	qdel(I)
+	var/gacha = rand(1,100)
+	if(gacha > 50)
+		playsound(src, 'sound/abnormalities//dreamingcurrent/dead.ogg', 80, TRUE, -3)
+		to_chat(user, "<span class='notice'>Nothing happens...</span>")
+		return
 
-//Gacha now locked in
+	switch(output)
+		if ("MONEY")
```
Comment (dsmith328) [inline code/modules/mob/living/simple_animal/abnormality/_tools/zayin/wishwell.dm:188]: According to coding standards, remove the space between the if-statement and it's condition, please.

**Review by dsmith328 — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/abnormality/_tools/teth/skin.dm (hunk 1 above)
Comment (EgorDinamit) [reply code/modules/mob/living/simple_animal/abnormality/_tools/teth/skin.dm:32]: Oh right, I was just copy-pasting what I wrote for behavior.

File: code/modules/mob/living/simple_animal/abnormality/_tools/teth/theresia.dm (hunk 2 above)
Comment (EgorDinamit) [reply code/modules/mob/living/simple_animal/abnormality/_tools/teth/theresia.dm:42]: Once per process, yes.

File: code/modules/mob/living/simple_animal/abnormality/_tools/zayin/wishwell.dm (hunk 3 above)
Comment (EgorDinamit) [reply code/modules/mob/living/simple_animal/abnormality/_tools/zayin/wishwell.dm:188]: Wishing well needs a whole damn rewrite that I don't want to do at the moment, but sure, I can do that.

**Review by Kitsunemitsu — COMMENTED** [review]
Could you give Theresia a drag slowdown so pepple don't move at max speed with it

**Comment by EgorDinamit** [comment]
All done

**Review by dsmith328 — APPROVED** [review]
Issues seem resolved to me

**Review by dsmith328 — APPROVED** [review]
👍

**Pull request merged**